compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1633,17 +1633,26 @@ convertExpr env0 e = do
         withTmArg env field' args $ \x1 args ->
             withTmArg env record' args $ \x2 args ->
                 pure (ERecUpd (fromTCon record') (mkField $ fsToText name) x2 x1, args)
-    -- NOTE(MH): We only inline `getField` for record types. Projections on
+    -- NOTE(MH, MA): We only inline `getField` for record types, and only when the
+    -- type actually contains a field of the given name. Projections on
```

Comment:
the problem was that the inlined `ERecProj`/`ERecUpd` wouldn't pass the LF typechecker, so it was impossible to produce a dar that made use of custom `HasField` instances with dot syntax. Notably, custom `HasField` instances worked (and still do) if `getField`/`setField` were used "indirectly", i.e. with the type arguments left as variables, since the inlining cases here only match when those are concrete types